packages/sources/s3reader/src/transport/csv.ts

```diff
@@ -0,0 +1,118 @@
+    this.s3Client = new S3Client({})
+  }
+
+  async backgroundHandler(context: EndpointContext<TransportTypes>, entries: RequestParams[]) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<BaseEndpointTypes['Response']>
+    try {
+      response = await this._handleRequest(param)
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequest(param: RequestParams): Promise<AdapterResponse<TransportTypes['Response']>> {
+    const { bucket, key, headerRow, resultField, matcherField, matcherValue } = param
+    const providerDataRequestedUnixMs = Date.now()
+    const csvFileAsStr = await getFileFromS3(this.s3Client, bucket, key)
+    const answer = this.parseCSV(csvFileAsStr, headerRow, matcherField, matcherValue, resultField)
+
+    return {
+      result: answer,
+      data: {
+        result: answer,
+      },
+      statusCode: 200,
+      timestamps: {
+        providerDataRequestedUnixMs,
+        providerDataReceivedUnixMs: Date.now(),
+        providerIndicatedTimeUnixMs: undefined,
+      },
+    }
+  }
+
+  // csvFileAsStr: CSV file as string as received from S3
+  // headerRow: 1-indexed row of the CSV file to use as the header row
+  // matcherField: field to match with `matcherValue` to find the answer row
+  // matcherValue: value of field `matcherField` used to find the answer row
+  // resultField: header field containing the answer in matcher row
+  parseCSV(
+    csvFileAsStr: string,
+    headerRow: number,
+    matcherField: string,
+    matcherValue: string,
+    resultField: string,
+  ): number {
+    // from_line is 1-indexed
+    // columns: true sets first line as object fields rather than 2d arrays
+    const parser = parse(csvFileAsStr, { columns: true, from_line: headerRow })
+
+    // validate CSV contains headers matcherField and resultField
+    if (!(matcherField in parser[0])) {
+      throw new Error(`CSV file does not contain column header ${matcherField}`)
+    }
+    if (!(resultField in parser[0])) {
+      throw new Error(`CSV file does not contain column header ${resultField}`)
+    }
+
+    // find correct row using matcher
+    const row = parser.find((row: { [x: string]: string }) => row[matcherField] == matcherValue)
+    if (!row) {
+      throw new Error(`CSV file does not contain row where ${matcherField} == ${matcherValue}`)
+    }
+    return row[resultField]
```

Comment:
I think `csv-parse` returns all values as strings, so we'll want to convert this to a number

Reply:
Handled in added commit